src/leveled_pclerk.erl

```diff
@@ -285,6 +296,18 @@ do_merge(KL1, KL2, SinkLevel, SinkB, RP, NewSQN, MaxSQN, OptsSST, Additions) ->
                             Additions ++ [Entry])
     end.
 
+-spec grooming_scorer(list(manifest_entry())) -> manifest_entry().
+grooming_scorer(Sample) ->
+    ScoringFun = 
+        fun(ME) ->
+            TombCount = leveled_sst:sst_gettombcount(ME#manifest_entry.owner),
+            {TombCount, ME}
+        end,
+    ScoredSample =
+        lists:reverse(lists:ukeysort(1, lists:map(ScoringFun, Sample))),
```

Comment:
Not sure whether lists:reverse and lists:ukeysort (traversing the list in N and  N * log(N) are expensive, but you could just do a fold, I guess.
```
grooming_scorer([ME  | MEs]) ->
    {Highest, BestME} =
         grooming_scorer(leveled_sst:sst_gettombcount(ME#manifest_entry.owner), ME, MEs),
         leveled_log:log("PC024", [HighestTC]),
         BestME.

grooming_scorer(HighestTC, BestME, []) ->
     {HighestTC, BestME};
grooming_scorer(HighestTC, BestME, [ME | MEs]) ->
    TombCount = leveled_sst:sst_gettombcount(ME#manifest_entry.owner),
    case TombCount > HighestTC of
        true -> grooming_scorer(TombCount, ME, MEs);
        false -> grooming_scorer(HighestTC, BestME, [ME | MEs])
   end.
```
linear in sample size.

Reply:
OK, thanks, gone with a corrected version of this